### DISCUSSION TITLE
Custom sources and sorting

### DISCUSSION THREAD

**Discussion by kringkaste** [craftcms/cms#12002] [Ideas] [answered]
Custom sources are sorted by dateCreated asc by default. It would be great if they would either sort according to the general default of Craft postDate desc or if you could specify a default sorting directly in the source definition.

As a further option, it would then be conceivable that a different sorting could also be specified for all other, i.e. native sources.

Background: In companies there are sometimes policies that require the clearing of the LocalStorage after closing the browser. This deletes the personal sorting configuration of Craft every time and the employees have to set it up again every morning.

**Comment by rynpsc** [comment]
I just had the same idea.

In 4.3 it's going to be possible to use relative dates when applying element conditions, making it possible to have sources set up to do things such as:

- List entries updated in last x
- List orders made over x days ago that haven't been dispatched
- List entries publishing in the next x 

Most of the time you are going to want that source to default to a particular order e.g dateUpdated, dateCreated and so on.

If this was made possible I'm not sure what should happen if a user were to change the sorting like so:

1. Create a source to be sorted by Date Created
2. Navigate to that source and change the sort order via the table column heading or the new "View" and choose another attribute, e.g. Title
3. Navigate to another source
4. Navigate back to the first source.

Should it revert to Date Created or the override, Title? For the two uses cases above reverting to the default makes more sense to me as the override was probably only temporary, but this would be different to the current order by behaviour which remembers your last selection.

I thinking something like this

![source](https://user-images.githubusercontent.com/5955881/192028818-25ee5bae-689b-473f-9f8d-708a6c44b211.png)

**Reply by rynpsc** [reply]
If it always returned to the users override perhaps there could there be some sort of indication in the UI that the default sort is being overridden?

That could solve another issue that our clients run into with Structures where they change the sort by, forget about it and then don't know why they can't reorder things.

**Reply by brandonkelly** [reply]
The “Use defaults” button in the View menu should help with that. It’s only visible when some customizations have been made, and could be a nice troubleshooting step to start recommending when the structure view isn’t working as expected.

**Comment by brandonkelly** [comment, marked as answer]
Added a new “Default Sort” setting for Craft 4.3 🎉  (#12057)

It’s available for both native and custom element sources.

![A Customize Sources modal opened up to a custom element source, with a spotlight on the new Default Sort setting](https://user-images.githubusercontent.com/47792/193951914-a26ecdff-1e9e-4d90-b8de-cb3089a4c432.png)

**Reply by kringkaste** [reply]
The customer loves it :dancing_women: Thank you very much!
